router/router.go

```diff
@@ -187,6 +197,49 @@ func (rt *HandleT) workerProcess(worker *workerT) {
 				if previousFailedJobID != job.JobID {
 					panic(fmt.Errorf("previousFailedJobID:%d != job.JobID:%d", previousFailedJobID, job.JobID))
 				}
+				// if the job has failed before, check for next retry time
```

Comment:
While we are here, do you think we need this block. I think we should remove this.

```
		//If sink is not enabled mark all jobs as waiting
		if !rt.isEnabled {
```